lib/go/thrift/json_protocol.go

```diff
@@ -314,11 +314,14 @@ func (p *TJSONProtocol) ReadMapBegin(ctx context.Context) (keyType TType, valueT
 	if err != nil {
 		return keyType, valueType, size, err
 	}
-	err = checkSizeForProtocol(int32(iSize), p.cfg)
+	size = int(iSize)
+
+	minElemSize := p.getMinSerializedSize(keyType) + p.getMinSerializedSize(valueType)
```

Comment:
for json map, there's at least another `+1` for the `:`, I think? not that that's super important though.

Reply:
ahh that's a good catch. Open to interpretation on how to fix this, I think I didn't want to deviate too much from existing code structure. This is tightening the lower bound. Because, at this time it also doesn't consider e.g. quoting/spacing.

Reply:
yea that's my "not that that's super important" part :)